dags/synapse-datasets-created-7-days.py

```diff
@@ -220,16 +221,52 @@ def push_results_to_synapse_table(
             synapseclient.Table(schema=SYNAPSE_RESULTS_TABLE, values=data)
         )
 
+    @task
+    def push_public_results_to_synapse_table(
+        entity_created: List[EntityCreated], **context
+    ) -> None:
+        """Push only the public results to a Synapse table."""
+        data = []
+        # convert context["params"]["backfill_date_time"] to date in same format as date.today()
+        today = (
+            date.today()
+            if not context["params"]["backfill"]
+            else datetime.strptime(
+                context["params"]["backfill_date_time"], "%Y-%m-%d  %H:%M:%S"
+            ).date()
+        )
+        # Filter for public datasets only
+        for row in entity_created:
+            if row.is_public:
+                data.append(
+                    [
+                        row.name,
+                        row.id,
+                        row.project_id,
+                        row.node_type,
+                        row.content_type,
+                        row.created_on,
+                        row.created_by,
+                        row.is_public,
+                        today,
+                    ]
+                )
+        syn_hook = SynapseHook(context["params"]["synapse_conn_id"])
+        syn_hook.client.store(
+            synapseclient.Table(schema=SYNAPSE_PUBLIC_RESULTS_TABLE, values=data)
+        )
```

Comment:
What version of the python client is being used in orca?  We may be able to use the new models if it's newer versions of SYNPY.